### ISSUE TITLE
Question about Project References for non-PCL proviers

### ISSUE THREAD

**Issue by tofutim** [ericsink/SQLitePCL.raw#128]
Hi Eric, I think the provider for net45, for example, references back to the core for net45. Would it work just as well to refer back to the PCL (e.g., profile111) core? Of course this would not work for things like win81.

Let's say you were targeting net45, ios_unified, macosx, android and uwp10, then there would only be one core needed, the profile111 one.

**Comment by tofutim** [comment]
SQLitePCLRaw.provider.xsqlite3.net45.csproj references SQLitePCLRaw.core.net45.csproj, but I'm thinking it could just as well reference SQLitePCLRaw.core.profile111.csproj

**Comment by tofutim** [comment]
Eric, I think there might be a bug in find_core. If you generate all the cores with 

```
        items_csproj.Add(config_csproj.create_core("net35"));
        items_csproj.Add(config_csproj.create_core("net40"));
        items_csproj.Add(config_csproj.create_core("net45"));
        items_csproj.Add(config_csproj.create_core("ios_unified"));
        items_csproj.Add(config_csproj.create_core("macos"));
        // TODO items_csproj.Add(config_csproj.create_core("watchos"));
        items_csproj.Add(config_csproj.create_core("android"));
        items_csproj.Add(config_csproj.create_core("win8"));
        items_csproj.Add(config_csproj.create_core("win81"));
        items_csproj.Add(config_csproj.create_core("wpa81"));
        items_csproj.Add(config_csproj.create_core("uwp10"));
        items_csproj.Add(config_csproj.create_core("profile111"));
        items_csproj.Add(config_csproj.create_core("profile136"));
        items_csproj.Add(config_csproj.create_core("profile259"));
        items_csproj.Add(config_csproj.create_core("netstandard10"));
        items_csproj.Add(config_csproj.create_core("netstandard11"));
```

it will never pickup the profile111 core even though maybe it should prefer it (at least I would prefer it). If you only generate the profile111 core, then it will do the right thing for net45. Since the first 'find' will always match it net45 to net45.

I would propose to prefer  profile111 or netstandard11, and add macosx, android, and ios to the list. Will send a PR proposal.

```
public static config_csproj find_core(string env)
	{
        config_csproj cfg = find("core", env);
        if (cfg == null)
        {
            // TODO need to find a core that is compatible with env
            // TODO this should be smarter
            switch (env)
            {
                case "net40":
                    cfg = find("core", "profile136");
                    break;
                case "net45":
                    cfg = find("core", "profile111");
                    break;
                case "win81":
                    cfg = find("core", "profile111");
                    break;
                default:
                    cfg = find("core", "netstandard11");
                    //cfg = find("core", "profile259");
                    break;
            }
        }
```

**Comment by ericsink** [comment]
I'm aware of this set of issues.  The code isn't exactly optimal, but I don't consider it a bug.  It's an area of the code I intend to clean up at some point.

When I added support for netstandard, I did so in an additive way.  I support netstandard, but I don't depend on anything about it.

At some point I will change SQLitePCL.raw to embrace the religion of netstandard.  When I do, I'll stop building so many variations of every assembly.
